core/src/com/itemshop/character/walking/PathingSystem.java

```diff
@@ -0,0 +1,169 @@
+package com.itemshop.character.walking;
+
+import com.badlogic.ashley.core.ComponentMapper;
+import com.badlogic.ashley.core.Engine;
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.ashley.core.Family;
+import com.badlogic.ashley.systems.IntervalIteratingSystem;
+import com.badlogic.ashley.utils.ImmutableArray;
+import com.itemshop.character.FacingDirectionComponent;
+import com.itemshop.character.walking.astar.AStarNode;
+import com.itemshop.character.walking.astar.AStarPathfinder;
+import com.itemshop.movement.Direction;
+import com.itemshop.movement.MovementTileTransitionComponent;
+import com.itemshop.movement.WalkableTileComponent;
+import com.itemshop.render.PositionComponent;
+
+/**
+ * Handles processing of entities which can walk.
+ */
+public class PathingSystem extends IntervalIteratingSystem {
+
+	/** Component mappers to get components from entities. */
+	private ComponentMapper<PositionComponent> positionMapper;
+	private ComponentMapper<PathComponent> pathMapper;
+	private ComponentMapper<FacingDirectionComponent> facingDirectionMapper;
+	private ComponentMapper<MovementTileTransitionComponent> tileTransitionMapper;
+	private ComponentMapper<WalkComponent> walkMapper;
+	private ComponentMapper<WalkableTileComponent> walkableTileMapper;
+
+	/** The path finding system family. */
+	private static Family family = Family.all(PositionComponent.class, FacingDirectionComponent.class)
+			.one(PathComponent.class, MovementTileTransitionComponent.class).get();
+
+	/** The transition step to decrement transition offsets by. */
+	private static float TRANSITION_STEP = 1f / 8f;
+
+	/** The positioned walkable tile entities. We need to know these in order to carry our path finding. */
+	ImmutableArray<Entity> positionedWalkableTileEntities;
+
+	/**
+	 * Constructs the pathing system instance.
+	 */
+	public PathingSystem() {
+		super(family, TRANSITION_STEP / 2);
+
+		// Create the componentMappers.
+		positionMapper        = ComponentMapper.getFor(PositionComponent.class);
+		pathMapper            = ComponentMapper.getFor(PathComponent.class);
+		facingDirectionMapper = ComponentMapper.getFor(FacingDirectionComponent.class);
+		tileTransitionMapper  = ComponentMapper.getFor(MovementTileTransitionComponent.class);
+		walkMapper            = ComponentMapper.getFor(WalkComponent.class);
+		walkableTileMapper    = ComponentMapper.getFor(WalkableTileComponent.class);
+	}
+
+	/**
+	 * Called when this system is added to the engine.
+	 * @param engine the engine
+	 */
+	public void addedToEngine(Engine engine) {
+		super.addedToEngine(engine);
+		positionedWalkableTileEntities = engine.getEntitiesFor(Family.all(PositionComponent.class, WalkableTileComponent.class).get());
+	}
+
+	@Override
+	protected void processEntity(Entity entity) {
```

Comment:
Personally I would split some of the different branches out into different functions to make it easier to read, you don't have to do it if you don't want to